---
title: "BGGN-213"
subtitle: "First Year Exam Questions for 2022/2023"
format: 
  pdf:
    toc: false
    thanks:  http://thegrantlab.org/bggn213/
    include-in-header:
      - hyperref
  html:
    toc: true
    toc-title: Sections
---

<!-- This sets up the PDF form question area style -->

\kvsetkeys{Field}{charsize=10pt, bordercolor=, backgroundcolor={.93 .93 .99}, color={0 0 1}, donotspellcheck}

\renewcommand\DefaultOptionsofText{multiline=true, width=\linewidth, height=1in}

<!-- End style setup -->

::: callout-note
## Instructions

Save this document to your computer and open it in a PDF viewer such as Preview (available on every mac) or Adobe Acrobat Reader ([free for PC and Linux](https://get.adobe.com/reader/)). Be sure to add your name and UC San Diego personal identification number (PID) and email below before answering all questions in the space provided.\
\

```{=tex}
\begin{tabular}{lll}
 Student Name &  UCSD PID  & UCSD Email \\
\TextField[name=Name, width=6cm, height=0.6cm]{} & \TextField[name=PID, width=3.5cm, height=0.6cm]{} & \TextField[name=Email, width=4.51cm, height=0.6cm]{}
\end{tabular}
```
:::

## Overview:

This open-book, open-notes test consists of 10 required questions and one optional bonus point question (question 11). The number of points for each question is indicated in square parenthesizes at the beginning of each question.

No communication (electronic or otherwise) with your fellow students regarding this test until after the due date.

Please remember to:

-   Download the PDF version and open in Preview (Mac) or Acrobat Reader (Windows).
-   Type all your answers directly in the space provided below each question.
-   Save and upload your completed test to [gradescope](https://www.gradescope.com/courses/538974).

**Good luck!**

## Test Questions:

Visit the following [webpage](https://bioboot.github.io/bggn213_W23/class-material/exam/00_sequence_listing.html) and download your student specific sequences.

> **N.B.** These sequence are unique for you and you must use your sequences to answer the following questions in the space provided.

\

**Q1.** \[1pt\] What protein do these sequences correspond to?

\TextField[name=Q1, height=1.2cm]{}

\

**Q2.** \[6pts\] What are the tumor specific mutations in this particular case (e.g. A130V)?

\TextField[name=Q2]{}

\

**Q3.** \[1pts\] Do your mutations cluster to any particular domain and if so give the name and PFAM id of this domain? Alternately note whether your protein is single domain and provide it's PFAM id (e.g. PF02196).

\TextField[name=Q3, height=1.2cm]{}

\

**Q4.** \[2pts\] Using the [NCI-GDC](https://portal.gdc.cancer.gov) list the observed top 2 missense mutations in this protein (amino acid substitutions)?

\TextField[name=Q4, height=1.2cm]{}

\

**Q5.** \[2pts\] What two TCGA projects have the most cases affected by mutations of this gene?

\TextField[name=Q5]{}

\

**Q6.** \[3pts\] List one RCSB PDB identifier with 100% identity to the wt_healthy sequence and detail the percent coverage of your query sequence for this known structure? Alternately, provide the most similar in sequence PDB structure along with it's percent identity, coverage and E-value. 

\TextField[name=Q6]{}

\


**Q7.**  \[10pts\] Using [AlphaFold notebook](https://colab.research.google.com/github/sokrypton/ColabFold/blob/main/AlphaFold2.ipynb) generate a structural model using the default parameters for your **mutant** sequence. 

> Note that this can take some time depending upon your sequence length. If your model is taking many hours to generate or your input sequence yields a "too many amino acids" (i.e. length) error you can focus on the main PFAM domain of interest (your answer to Q3 above). 

Once complete save the resulting PDB format file for your records and use [Mol-star](https://molstar.org/viewer/) (or your favorite molecular viewer) to render a molecular figure. In this figure please clearly show your mutant amino acid **side chains as spacefill** and the protein as **cartoon colored by local alpha fold pLDDT quality score**. This score is contained in the B-factor column of your PDB downloaded file. **Upload this image to [GradeScope](https://www.gradescope.com/courses/538974)**. 

\

**Q8.** \[2pts\] Considering only your mutations in high quality structure regions (with a pLDDT score > 70) are any of the mutations on the surface of the protein and hence have a potential to interfere with protein-protein interaction events? List these mutations below (e.g. A130V)

\TextField[name=Q8]{}

\


**Q9.** \[5pts\] Please comment on how useful and/or reliable you think your AlphaFold structural model is for your entire sequence and the main domain where your mutations lie? You may wish to compare your model to the PDB structure you found in Q6. 

\TextField[name=Q9, height=5cm]{}

\


**Q10.** \[10pts\] Obtain the most recently dated COVID-19 Variant Data from the [California Health and Human Services (CHHS) open data site](https://data.chhs.ca.gov/dataset/covid-19-variant-data).  


Upload to [gradescope](https://www.gradescope.com/courses/538974) a PDF format report generated from a Quarto or Rmarkdown document that
demonstrates reading the above CSV file and generating the below visualization of this data.

![Example plot for Q10.](plot.png)

> **NB.** You can chose how to make this plot and whether you want to make improvements or stylistic changes. However, you are strongly encouraged to use the ggplot2, lubridate and dplyr packages for this task. Please make sure your name and PID number is on the first page and that your report contains all of your code, text description/narrative text of why you doing a particular task/code chunk and the resulting figure.

\


**Q11.** \[10pts\]  *Optional: This is not a required question but will yield you 10 extra bonus points*. Using git upload (a.k.a. push) your RStudio project containing your complete work for Q10 to GitHub and provide a link to your project directory here:

\TextField[name=Q11, height=1.2cm]{}

\

# - End of Test -